### PR TITLE
Enable Controlled Legend working in Declarative HBC

### PR DESCRIPTION
**Pull request by Anush2303**
## Previous Behavior

Controlled Legends not working in Declarative HBC.

## New Behavior

Controlled legends working in declarative HBC. 

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: change/@fluentui-react-charting-f8073f84-d26a-4b28-992e-379202a63f5d.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-f8073f84-d26a-4b28-992e-379202a63f5d.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33476&commitId=26f00aeaa43eb76c1947cb5598b13042defedf07&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.No callout width specified.chromium.png|2319| Changed|
|Callout.Top left edge.chromium.png|2307| Changed|
|Callout.Bottom Left Edge RTL.chromium.png|2178| Changed|

</details>


  <details><summary><b>Keytip</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Keytip.Offset.chromium.png|121| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
       color: '',
       dataForHoverCard: 0,
       isCalloutVisible: false,
-      isLegendSelected: false,
+      isLegendSelected: props.legendProps?.selectedLegend?.length! > 0,
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:87]: What if the legend title text is '' (empty string)? In that case even if the legend is selected, isLegendSelected will be false

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:87]: if legend title text is empty string, that means no legend is selected.

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:87]: Yes, correct.

**Review by AtishayMsft — APPROVED** [review]
:shipit:

**Pull request merged**